R/readSource.R

```diff
@@ -4,41 +4,41 @@
 #' wrapper for specific functions designed for the different possible source
 #' types.
 #'
-#' @param type source type, e.g. "IEA". A list of all available source types
-#' can be retrieved with function \code{\link{getSources}}.
-#' @param subtype For some sources there are subtypes of the source, for these
-#' source the subtype can be specified with this argument. If a source does not
+#' @param type A character string referring to the source type, e.g. "IEA" which would
+#' internally call a function called `readIEA` (the "wrapped function"). A list of
+#' available source types can be retrieved with function \code{\link{getSources}}.
+#' @param subtype A character string. For some sources there are subtypes of the source, for these
+#' sources the subtype can be specified with this argument. If a source does not
 #' have subtypes, subtypes should not be set.
-#' @param subset Similar to \code{subtype} a source can also have \code{subsets}. A \code{subsets}
-#' can be used to only read part of the data. This can in particular make sense for huge
-#' data sets where reading in the whole data set might be impractical or even
-#' infeasible.
+#' @param subset A character string. Similar to \code{subtype} a source can also have \code{subsets}.
+#' A \code{subsets} can be used to only read part of the data. This can in particular make sense
+#' for huge data sets where reading in the whole data set might be impractical or even infeasible.
 #' @param convert Boolean indicating whether input data conversion to
 #' ISO countries should be done or not. In addition it can be set to "onlycorrect"
 #' for sources with a separate correctXXX-function.
-#' @return magpie object with the temporal and data dimensionality of the
-#' source data. Spatial will either agree with the source data or will be on
-#' ISO code country level depending on your choice for the argument "convert".
-#' @author Jan Philipp Dietrich, Anastasis Giannousakis, Lavinia Baumstark
+#' @return The read-in data, usually a magpie object. Technically speaking
+#' `if (is.list(result)) result$x else result` will be returned, where `result` is the
+#' return value of the wrapped function. The temporal and data dimensionality
+#' should match the source data. The spatial dimension should either match the source data or,
+#' if the convert argument is set to TRUE, should be on ISO code country level. For magpie objects
+#' magclass::clean_magpie is run and if convert = TRUE ISO code country level is checked.
```

Comment:
why do you mention `clean_magpie` here? If you want to add more details I would add them as a note, not as part of the description of the return value

Reply:
You mean as `vcat(2, ...)`? `clean_magpie` changes the return value, so I think it makes sense here, no?

Reply:
no, I meant in the documentation. There is a category `@details` which could be used for it (I thought it was `@note` which is why my wording was misleading, sorry for that).

Reply:
ok sure